packages/nimble-components/src/select/specs/built-in-labels.md

```diff
@@ -0,0 +1,77 @@
+        <nimble-list-option>B</nimble-list-option>
+    </nimble-select>
+    ```
+
+### Templates / Styling
+
+Template updates:
+
+-   A new `label` element will be added at the root of the control template, which will reflect the default slotted content (minus the list options which are already handled elsewhere)
+-   A new container `div` will be added at the same level, to contain the control part (`div[part="control"]`)
+-   The element host will become `inline-block` so the label is laid out correctly. (Formerly `inline-flex`, which will move to the container `div`)
+
+Currently our shared dropdown styling (`patterns/dropdown/styles.ts`) applies focus/error-visible styling directly on `::host` - most of those styles will instead target the container `div`.
+
+### Accessibility / ARIA
+
+`nimble-combobox`: Currently has 2 ARIA/accessibility issues:
+
+-   Elements must only use supported ARIA attributes: `nimble-combobox` `nimble-toggle-button`: ARIA attribute is not allowed: `aria-expanded="false"`
+    -   Can be resolved via `aria-hidden="true"` on the dropdown toggle button, as it's not keyboard-focusable.
+-   ARIA commands must have an accessible name: `nimble-combobox` `nimble-toggle-button` `div[role="button"]`
+    -   Will be resolved by adding the `label` element
+
+`nimble-select`: Currently has 1 ARIA/accessibility issue:
+
+-   ARIA input fields must have an accessible name: `nimble-select`
+    -   The `label` element is not sufficient to resolve this, because it's in the shadow tree, whereas the ARIA role is on the `nimble-select` itself (unlike the combobox).
+    -   Will be resolved by adding the `aria-label` attribute to `nimble-select`, which will be set to the same label text shown in the `label` element. (`nimble-list-option-group` [does something similar](https://github.com/ni/nimble/blob/4948838ca895a04d7ebe16f595176b2050863f15/packages/nimble-components/src/list-option-group/template.ts#L14))
+
+### Impact to nimble-rich-text-mention-listbox
+
+`nimble-rich-text-mention-listbox` will not support this feature, meaning it will not contain a `label` element in its template. The current plan is to not address any ARIA label issues on it either (see Open Issues).  
+However, since we have shared dropdown styling, its template may need minor updates (e.g. to add the same parent/container `div` as select/combobox are getting), but we'll ensure it has the same visual appearance as before.
+
+## Alternative Implementations / Designs
+
+**Label API:** We have several different paradigms for specifying label text in Nimble, in addition to the default slot:
+
+-   A `label` slot, with the expectation clients will add a `label` element in it (`nimble-radio-group`)
+-   A `label` attribute (`nimble-option-group`, though text content in the default slot is also supported)
+
+The default slot approach was chosen for this feature as it's the most common in Nimble, and the simplest to use.
+
+## Open Issues
```

Comment:
Not sure if this is the place to figure this out (especially with Fred about to be OOO) but do we know what usage guidance we'll have for setting labels? Should it always be provided? Only provided if the purpose of the control isn't obvious from its placeholder / default value? Do we need a way to provide it for a screen reader but not show it visibly?

Reply:
Good questions. Don't think we need to block this HLD on the answers, so added a task in the linked issue about docs/ usage guidance, and captured your questions there.